bin/git-create-branch

```diff
@@ -18,23 +26,34 @@ do
 	shift
 done
 
-if [[ -z $BRANCH ]] && [[ -n $REMOTE ]] 
+# handle ambiguous `-r` option argument by shift
+if [[ -z $BRANCH ]] && [[ -n $REMOTE ]]
 then
 	BRANCH=$REMOTE
 	REMOTE=origin
 fi
 
-test -z $BRANCH && echo "branch required." 1>&2 && exit 1
+test -z $BRANCH && echo "branch argument required." 1>&2 && exit 1
+
 if [[ -n $REMOTE ]]
 then
-	git ls-remote --exit-code $REMOTE &>/dev/null
-	if [ $? = 0 ]
+	git ls-remote --exit-code $REMOTE 1>/dev/null 2>/tmp/ls-remote-error
+	REMOTE_EXIT=$?
+	REMOTE_ERROR=$(</tmp/ls-remote-error)
+	rm -f /tmp/ls-remote-error
+	if [ $REMOTE_EXIT -eq 0 ]
 	then
 		git push $REMOTE HEAD:refs/heads/$BRANCH
 		git fetch $REMOTE
 		git checkout --track -b $BRANCH $REMOTE/$BRANCH
 		exit $?
+	else
+		echo
+		echo "    Error connecting to remote '$REMOTE': $REMOTE_ERROR"
+		echo
+		exit $REMOTE_EXIT
 	fi
 fi
 
 git checkout -b $BRANCH
+exit $?
```

Comment:
Is this line useful? The shell script always exits with the exit code of the last command.

Reply:
:blush: